Actions/DetermineDeploymentEnvironments/DetermineDeploymentEnvironments.ps1

```diff
@@ -0,0 +1,177 @@
+
+$deploymentEnvironments = @{}
+
+if (!($environments)) {
+    # If no environments are defined and the user specified a single environment, use that environment
+    # This allows the user to specify a single environment without having to define it in the settings
+    if ($getenvironments -notcontains '*' -and $getenvironments -notcontains '?' -and $getenvironments -notcontains ',') {
+        $envName = $getEnvironments.Split(' ')[0]
+        $deploymentEnvironments += @{
+            "$getEnvironments" = @{
+                "EnvironmentName" = $envName
+                "Branches" = $null
+                "BranchesFromPolicy" = $null
+                "Projects" = '*'
+                "ForceSync" = $false
+                "ContinuousDeployment" = !($getEnvironments -like '* (PROD)' -or $getEnvironments -like '* (Production)' -or $getEnvironments -like '* (FAT)' -or $getEnvironments -like '* (Final Acceptance Test)')
+                "runs-on" = @($settings."runs-on".Split(',').Trim())
+            }
+        }
+    }
+}
+else {
+    $environments | ForEach-Object {
+        $environmentName = $_
+        Write-Host "Environment: $environmentName"
+        $envName = $_.Split(' ')[0]
+
+        # Check Obsolete Settings
+        "$($envName)-Projects","$($envName)_Projects" | ForEach-Object {
+            if ($settings.Contains($_)) {
+                throw "The setting $_ is obsolete and should be replaced by using the Projects property in the DeployTo$envName setting in .github/AL-Go-Settings.json instead"
+            }
+        }
+
+        # Default Deployment settings are:
+        # - environment name: same
+        # - branches: main
+        # - projects: all
+        # - continuous deployment: only for environments not tagged with PROD or FAT
+        # - runs-on: same as settings."runs-on"
+        $deploymentSettings = @{
+            "EnvironmentType" = "SaaS"
+            "EnvironmentName" = $envName
+            "Branches" = $null
+            "BranchesFromPolicy" = $null
+            "Projects" = '*'
+            "ForceSync" = $false
+            "ContinuousDeployment" = $null
+            "runs-on" = @($settings."runs-on".Split(',').Trim())
+        }
+
+        # Check DeployTo<environmentName> setting
+        $settingsName = "DeployTo$envName"
+        if ($settings.ContainsKey($settingsName)) {
+            # If a DeployTo<environmentName> setting exists - use values from this (over the defaults)
+            $deployTo = $settings."$settingsName"
+            'EnvironmentType','EnvironmentName','Branches','Projects','ForceSync','ContinuousDeployment','runs-on' | ForEach-Object {
+                if ($deployTo.ContainsKey($_)) {
+                    $deploymentSettings."$_" = $deployTo."$_"
+                }
+            }
+        }
+
+        # Get Branch policies on GitHub Environment
+        $ghEnvironment = $ghEnvironments | Where-Object { $_.name -eq $environmentName }
```

Comment:
Suggestion: Consider refactoring this block (Get-BranchPolicies or something like that)

Reply:
Agree